Source/FakeItEasy/Core/DefaultFakeManagerAccessor.cs

```diff
@@ -23,11 +22,20 @@ public FakeManager GetFakeManager(object proxy)
 
             var taggable = AsTaggable(proxy);
 
+            if (taggable.Tag == null)
+            {
+                throw new ArgumentException("The specified object does not have a FakeManager attached.");

```

Comment:
I'm wondering how much this adds. Is this the message that will be surfaced to the end user?

My concern here is that `taggable.Tag == null` and "The specified object does not have a FakeManager attached." do not have any intuitive connection, and I'm wondering whether the message will mean anything at all to the end user.